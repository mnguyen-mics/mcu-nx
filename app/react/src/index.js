import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';

import configureStore from './store';
import { Navigator } from './containers/Navigator';

const store = configureStore();

const micsProvider = (
  <Provider store={store}>
    <Navigator />
  </Provider>
);

render(
  micsProvider,
  document.getElementById('mcs-react-app') // eslint-disable-line no-undef
);
