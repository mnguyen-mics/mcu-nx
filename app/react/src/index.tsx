/* eslint-disable */
import React from 'react';
import { render } from 'react-dom';

import App from './App';

declare global {
  interface Window {
    require: any;
    mainRev: any;
  }
}

render(<App />, document.getElementById('mcs-react-app'), () => {
  window.require([window.mainRev], () => {
    //
  });
});
