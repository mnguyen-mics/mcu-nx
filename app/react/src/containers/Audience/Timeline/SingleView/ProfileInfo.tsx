import * as React from 'react';
import cuid from 'cuid';
import { Row, Col, Tooltip } from 'antd';
import { FormattedMessage } from 'react-intl';
import messages from '../messages';

export interface ProfileInfoProps {
  profile?: any
}

interface State {
  showMore: boolean;
}

export default class ProfileInfo extends React.Component<ProfileInfoProps, State> {

  static defaultProps = {
    profile: {}
  }

  constructor(props: ProfileInfoProps) {
    super(props);
    this.state =Â {
      showMore: false,
    }
  }

  public render() {
    const { profile } = this.props;
    
    const convertedObjectToArray = Object.keys(profile).map(key => {
      return [key, profile[key]];
    });

    const profileFormatted =
      convertedObjectToArray.length > 5 && !this.state.showMore
        ? convertedObjectToArray.slice(0, 5)
        : convertedObjectToArray;

    const canViewMore = convertedObjectToArray.length > 5 ? true : false;

    const onViewMoreClick = (e: any) => {
      e.preventDefault();
      this.setState({ showMore: true });
    };

    const onViewLessClick = (e: any) => {
      e.preventDefault();
      this.setState({ showMore: false });
    };

    const hasItems = !!Object.keys(profile).length;

    const generateValues = (t: object) => Object.keys(t).map(k => [k, (t as any)[k]])

    const generateItems = (profileInfoSlice: [string, any], shouldSlide: boolean = false, margin: number = -5): React.ReactNode => {
      if (typeof profileInfoSlice[1] === 'string' || typeof profileInfoSlice[1] === 'number' || !profileInfoSlice[1]) {
        return (<Row gutter={10} key={cuid()} className={"table-line"} style={{ marginLeft: shouldSlide ? margin + 10 : margin }}>
        <Col className="table-left" span={12}>
          <Tooltip title={profileInfoSlice[0]}>{profileInfoSlice[0]}</Tooltip>
        </Col>
        <Col className="table-right" span={12}>
          <Tooltip title={profileInfoSlice[1] ? profileInfoSlice[1] : 'null'}>{profileInfoSlice[1] ? profileInfoSlice[1] : 'null'}</Tooltip>
        </Col>
      </Row>)
      }

      if (Array.isArray(profileInfoSlice[1])) {
        return  (<Row gutter={10} key={cuid()} className={"table-line"} style={{ marginLeft: shouldSlide ? margin + 10 : margin }}>
          <Col className="table-left" span={12}>
            <Tooltip title={profileInfoSlice[0]}>{profileInfoSlice[0]}</Tooltip>
          </Col>
          <Col className="table-right" span={12}>
            <Tooltip title={JSON.stringify(profileInfoSlice[1])}>{JSON.stringify(profileInfoSlice[1])}</Tooltip>
          </Col>
        </Row>)
      }

      return (
        <Row gutter={10} key={cuid()} className={"table-line"} style={{ marginLeft: shouldSlide ? margin + 10 : margin }}>
          <Col className="table-left" span={24}>
            <Tooltip title={profileInfoSlice[0]}>{profileInfoSlice[0]}:</Tooltip>
          </Col>
          <Col className={'p-l-10'} span={24}>
            {generateValues(profileInfoSlice[1] as any).map(item => {
              return generateItems(item as [string, any], true, margin + 10)
            })}
          </Col>
        </Row>
      )
    }

    return (
      <div>
        {profileFormatted &&
          profileFormatted.map(profil => {
            return generateItems(profil as any)
          })}
        {(profileFormatted.length === 0 || hasItems === false) && (
          <span>
            <FormattedMessage {...messages.emptyProfile} />
          </span>
        )}
        {canViewMore ? (
          !this.state.showMore ? (
            <div className="mcs-card-footer">
              <button
                className="mcs-card-footer-link"
                onClick={onViewMoreClick}
              >
                <FormattedMessage {...messages.viewMore} />
              </button>
            </div>
          ) : (
            <div className="mcs-card-footer">
              <button
                className="mcs-card-footer-link"
                onClick={onViewLessClick}
              >
                <FormattedMessage {...messages.viewLess} />
              </button>
            </div>
          )
        ) : null}
      </div>
    );
  }
}
