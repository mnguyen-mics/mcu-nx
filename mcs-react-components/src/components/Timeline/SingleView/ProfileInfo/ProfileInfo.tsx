import * as React from 'react';
import * as _cuid from 'cuid';
import { Row, Col, Tooltip } from 'antd';
import { FormattedMessage, defineMessages } from 'react-intl';

const cuid = _cuid;

const messages = defineMessages({
  viewMore: {
    id: 'audience.monitoring.timeline.content.viewMore',
    defaultMessage: 'View More',
  },
  viewLess: {
    id: 'audience.monitoring.timeline.content.viewLess',
    defaultMessage: 'View Less',
  },
  emptyProfile: {
    id: 'audience.monitoring.audience.monitoring.timeline.card.profile.empty',
    defaultMessage: 'This user has no Profile Information',
  },
});

export interface ProfileInfoProps {
  profile?: any;
}

interface State {
  showMore: boolean;
}

export default class ProfileInfo extends React.Component<
  ProfileInfoProps,
  State
> {
  static defaultProps = {
    profile: {},
  };

  constructor(props: ProfileInfoProps) {
    super(props);
    this.state = {
      showMore: false,
    };
  }

  public render() {
    const { profile } = this.props;

    const convertedObjectToArray = Object.keys(profile).map(key => {
      return [key, profile[key]];
    });

    const profileFormatted =
      convertedObjectToArray.length > 5 && !this.state.showMore
        ? convertedObjectToArray.slice(0, 5)
        : convertedObjectToArray;

    const canViewMore = convertedObjectToArray.length > 5 ? true : false;

    const onViewMoreClick = (e: any) => {
      e.preventDefault();
      this.setState({ showMore: true });
    };

    const onViewLessClick = (e: any) => {
      e.preventDefault();
      this.setState({ showMore: false });
    };

    const hasItems = !!Object.keys(profile).length;

    const generateValues = (t: object) => {
      return Object.keys(t).map(k => [k, (t as any)[k]]);
    };

    const generateItems = (
      profileInfoSlice: [string, any],
      shouldSlide: boolean = false,
      margin: number = -5,
    ): React.ReactNode => {
      if (
        typeof profileInfoSlice[1] === 'string' ||
        typeof profileInfoSlice[1] === 'number' ||
        typeof profileInfoSlice[1] === 'boolean'
      ) {
        const filteredInfoValue = String(profileInfoSlice[1]) || 'empty';
        return (
          <Row
            gutter={10}
            key={cuid()}
            className={'table-line'}
            style={{ marginLeft: shouldSlide ? margin + 10 : margin }}
          >
            <Col className="table-left" span={12}>
              <Tooltip title={profileInfoSlice[0]}>
                {profileInfoSlice[0]}
              </Tooltip>
            </Col>
            <Col className="table-right" span={12}>
              <Tooltip title={filteredInfoValue}>{filteredInfoValue}</Tooltip>
            </Col>
          </Row>
        );
      }

      if (Array.isArray(profileInfoSlice[1])) {
        return (
          <Row
            gutter={10}
            key={cuid()}
            className={'table-line'}
            style={{ marginLeft: shouldSlide ? margin + 10 : margin }}
          >
            <Col className="table-left" span={12}>
              <Tooltip title={profileInfoSlice[0]}>
                {profileInfoSlice[0]}
              </Tooltip>
            </Col>
            <Col className="table-right" span={12}>
              <Tooltip title={JSON.stringify(profileInfoSlice[1])}>
                {JSON.stringify(profileInfoSlice[1])}
              </Tooltip>
            </Col>
          </Row>
        );
      }

      if (!profileInfoSlice[1]) {
        return (
          <Row
            gutter={10}
            key={cuid()}
            className={'table-line'}
            style={{ marginLeft: shouldSlide ? margin + 10 : margin }}
          >
            <Col className="table-left" span={12}>
              <Tooltip title={profileInfoSlice[0]}>
                {profileInfoSlice[0]}:
              </Tooltip>
            </Col>
            <Col className={'p-l-10'} span={12}>
              <i>{JSON.stringify(profileInfoSlice[1])}</i>
            </Col>
          </Row>
        );
      }

      return (
        <Row
          gutter={10}
          key={cuid()}
          className={'table-line'}
          style={{ marginLeft: shouldSlide ? margin + 10 : margin }}
        >
          <Col className="table-left" span={24}>
            <Tooltip title={profileInfoSlice[0]}>
              {profileInfoSlice[0]}:
            </Tooltip>
          </Col>
          <Col className={'p-l-10'} span={24}>
            {generateValues(profileInfoSlice[1] as any).map(item => {
              return generateItems(item as [string, any], true, margin + 10);
            })}
          </Col>
        </Row>
      );
    };

    return (
      <div>
        {profileFormatted &&
          profileFormatted.map(profil => {
            return generateItems(profil as any);
          })}
        {(profileFormatted.length === 0 || hasItems === false) && (
          <span>
            <FormattedMessage {...messages.emptyProfile} />
          </span>
        )}
        {canViewMore ? (
          !this.state.showMore ? (
            <div className="mcs-card-footer">
              <button
                className="mcs-card-footer-link"
                onClick={onViewMoreClick}
              >
                <FormattedMessage {...messages.viewMore} />
              </button>
            </div>
          ) : (
            <div className="mcs-card-footer">
              <button
                className="mcs-card-footer-link"
                onClick={onViewLessClick}
              >
                <FormattedMessage {...messages.viewLess} />
              </button>
            </div>
          )
        ) : null}
      </div>
    );
  }
}
