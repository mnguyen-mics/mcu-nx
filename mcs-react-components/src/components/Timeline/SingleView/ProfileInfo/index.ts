import ProfileInfo from './ProfileInfo';

export { ProfileInfoProps } from './ProfileInfo';

export default ProfileInfo;
