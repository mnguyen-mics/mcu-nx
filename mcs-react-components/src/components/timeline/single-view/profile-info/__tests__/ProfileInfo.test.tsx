import 'jest';
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import ProfileInfo, { ProfileInfoProps } from '../ProfileInfo';
import { IntlProvider } from 'react-intl';

it('renders the ProfileInfo', () => {
  const props: ProfileInfoProps = {
    profile: {
      $compartment_id: '1196',
      $creation_ts: 1533207442669,
      $items: [],
      $last_modified_ts: 1555517236970,
      $user_account_id: null,
      channel_id: '2748',
      usr_age: '',
      usr_gender: 'M.',
      usr_guid: '',
      usr_postalcode: '',
      usr_segment: '',
    },
  };
  const component = TestRenderer.create(
    <IntlProvider locale='en'>
      <ProfileInfo {...props} />
    </IntlProvider>,
  );
  const res = component.toJSON();
  expect(res).toMatchSnapshot();
});
