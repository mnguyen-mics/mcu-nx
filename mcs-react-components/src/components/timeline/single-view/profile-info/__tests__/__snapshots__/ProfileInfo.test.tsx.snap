// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ProfileInfo 1`] = `
<div
  className="mcs-profile-info"
>
  <div
    className="ant-row table-line"
    style={
      Object {
        "marginLeft": -5,
        "marginRight": -5,
      }
    }
  >
    <div
      className="ant-col-12 table-left"
      style={
        Object {
          "paddingLeft": 5,
          "paddingRight": 5,
        }
      }
    >
      <span
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        $compartment_id
      </span>
    </div>
    <div
      className="ant-col-12 table-right"
      style={
        Object {
          "paddingLeft": 5,
          "paddingRight": 5,
        }
      }
    >
      <span
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        1196
      </span>
    </div>
  </div>
  <div
    className="ant-row table-line"
    style={
      Object {
        "marginLeft": -5,
        "marginRight": -5,
      }
    }
  >
    <div
      className="ant-col-12 table-left"
      style={
        Object {
          "paddingLeft": 5,
          "paddingRight": 5,
        }
      }
    >
      <span
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        $creation_ts
      </span>
    </div>
    <div
      className="ant-col-12 table-right"
      style={
        Object {
          "paddingLeft": 5,
          "paddingRight": 5,
        }
      }
    >
      <span
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        1533207442669
      </span>
    </div>
  </div>
  <div
    className="ant-row table-line"
    style={
      Object {
        "marginLeft": -5,
        "marginRight": -5,
      }
    }
  >
    <div
      className="ant-col-12 table-left"
      style={
        Object {
          "paddingLeft": 5,
          "paddingRight": 5,
        }
      }
    >
      <span
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        $items
      </span>
    </div>
    <div
      className="ant-col-12 table-right"
      style={
        Object {
          "paddingLeft": 5,
          "paddingRight": 5,
        }
      }
    >
      <span
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        []
      </span>
    </div>
  </div>
  <div
    className="ant-row table-line"
    style={
      Object {
        "marginLeft": -5,
        "marginRight": -5,
      }
    }
  >
    <div
      className="ant-col-12 table-left"
      style={
        Object {
          "paddingLeft": 5,
          "paddingRight": 5,
        }
      }
    >
      <span
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        $last_modified_ts
      </span>
    </div>
    <div
      className="ant-col-12 table-right"
      style={
        Object {
          "paddingLeft": 5,
          "paddingRight": 5,
        }
      }
    >
      <span
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        1555517236970
      </span>
    </div>
  </div>
  <div
    className="ant-row table-line"
    style={
      Object {
        "marginLeft": -5,
        "marginRight": -5,
      }
    }
  >
    <div
      className="ant-col-12 table-left"
      style={
        Object {
          "paddingLeft": 5,
          "paddingRight": 5,
        }
      }
    >
      <span
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        $user_account_id
        :
      </span>
    </div>
    <div
      className="ant-col-12 p-l-10"
      style={
        Object {
          "paddingLeft": 5,
          "paddingRight": 5,
        }
      }
    >
      <i>
        null
      </i>
    </div>
  </div>
  <div
    className="mcs-card-footer"
  >
    <button
      className="mcs-card-footer-link"
      onClick={[Function]}
    >
      <span>
        View More
      </span>
    </button>
  </div>
</div>
`;
